sdk/python/feast/feature_server.py

```diff
@@ -1,16 +1,19 @@
 import json
 import sys
 import threading
+import time
 import traceback
 from contextlib import asynccontextmanager
 from typing import List, Optional
 
 import pandas as pd
+import psutil
```

Comment:
These shouldn't be top-level imports here. The goal is to have both `psutil` and `prometheus_client` as optional dependencies, right? 

Maybe best to most of the changes from here to a new `metrics.py` file and import them only during `start_server` after checking `metrics` boolean flag first.